File: Java/Java source-code/quantile.java


public class quantile {
    public static double quantile(Double[] sortedElements, Double phi) {
        int n = sortedElements.length;
        double index = phi * (n - 1);
        int lhs = (int) index;
        double delta = index - lhs;
        double result;
        if (n == 0) {
            return 0.0;
        }
        if (lhs == n - 1) {
            result = sortedElements[lhs];
        } else {
            result = (1 - delta) * sortedElements[lhs] + delta * sortedElements[lhs + 1];
        }
        return result;
    }
}

File: Methods-Python/quantile.py
def quantile(sortedElements, phi):

    n = len(sortedElements)
    index = phi * (n - 1)
    lhs = int(index)

    delta = index - lhs

    if n == 0:
        return 0

    if lhs == n - 1:
        result = sortedElements[lhs]

    else:
        result = (1 - delta) * sortedElements[lhs] + delta * sortedElements[lhs + 1]

    return result
